src/platform_impl/web/web_sys/timeout.rs

```diff
@@ -73,12 +91,34 @@ impl AnimationFrameRequest {
     }
 }
 
-impl Drop for AnimationFrameRequest {
+impl Drop for IdleCallback {
     fn drop(&mut self) {
         if !(*self.fired).get() {
-            self.window
-                .cancel_animation_frame(self.handle)
-                .expect("Failed to cancel animation frame");
+            match self.handle {
+                Handle::IdleCallback(handle) => self.window.cancel_idle_callback(handle),
+                Handle::Timeout(handle) => self.window.clear_timeout_with_handle(handle),
+            }
         }
     }
 }
+
+fn has_idle_callback_support(window: &web_sys::Window) -> bool {
+    thread_local! {
+        static IDLE_CALLBACK_SUPPORT: OnceCell<bool> = OnceCell::new();
+    }
```

Comment:
The thread local here might be more expensive than just an atomic variable.

Reply:
What, why? I've been using this a lot until now.

Reply:
I learned a lot going down this rabbit hole, things I honestly didn't want to know.

In any case, in this case `sync::OnceCell` is more expensive then the `unsync` version.